cepheus-cep/src/main/java/com/orange/cepheus/cep/model/Provider.java

```diff
@@ -96,4 +96,18 @@ public String getServicePath() {
     public void setServicePath(String servicePath) {
         this.servicePath = servicePath;
     }
+         
+   public boolean hasSameOrigin(Provider otherProvider) {
+		if (otherProvider != null) {
+			if (otherProvider.url != null && !otherProvider.url.isEmpty()) {
+				return url.equals(otherProvider.url)
+						&& (servicePath == null || servicePath
```

Comment:
With this code, you are missing a case: if `servicePath` is `null`, you are considering that the `otherProvider` has the same origin whatever is in `otherProvider.servicePath` which is not correct (its also the same for serviceName on the next line)...

So the correct code shoud be :
`return url.equals(otherProvider.url) && servicePath == otherProvider.servicePath && serviceName == otherServiceName`.

This ensures that if no servicePath / serviceName are defined, we still match the otherProviders that only have with no servicePath / serviceName. 

It might be good to add some tests to test that the behavior is correct.

Reply:
Hi @marcc-orange,

I have understood the missing case condition you have stated and am looking forward to it. Also,the condition in return statement-> (this.servicePath == otherProvider.servicePath) always return false because whenever a post request for the same configuration is sent, a new object is created for the (this.servicepath) and hence the return statement returns false.(Similarly,for serviceName also)

Thus,whenever we are sending the same configuration again consecutively, it evaluates to false and a new subscription is made for the same provider values.